tests/unit/moduleapi/blockonkeys.tcl

```diff
@@ -286,4 +286,21 @@ start_server {tags {"modules"}} {
         assert_equal {gg ff ee dd cc} [$rd read]
         $rd close
     }
+    
+    test {Module explicit unblock when blocked on keys} {
```

Comment:
@ranshid maybe we can test it using a more realistic scenario? e.g. CLIENT UNBLOCK

using RedisModule_UnblockClient inside the reply callback is a gross misuse of the API

Reply:
@guybe7 that would have been my first choice. but CLIENT UNBLOCK is already safe (removing the CLIENT_PENDING_COMMAND)